haystack/preview/components/preprocessors/text_document_splitter.py

```diff
@@ -43,8 +43,9 @@ def run(self, documents: List[Document]):
         :return: A list of documents with the split texts.
         """
 
-        if not documents or not isinstance(documents, list) or not isinstance(documents[0], Document):
+        if not isinstance(documents, list) or (len(documents) > 0 and not isinstance(documents[0], Document)):
```

Comment:
Let's use `documents` instead of `len(documents) > 0` here. So the line of code would be:
`if not isinstance(documents, list) or documents and not isinstance(documents[0], Document):`